aws/resource_aws_elasticache_global_replication_group.go

```diff
@@ -0,0 +1,303 @@
+	}
+
+	if v, ok := d.GetOk("global_replication_group_description"); ok {
+		input.GlobalReplicationGroupDescription = aws.String(v.(string))
+	}
+
+	output, err := conn.CreateGlobalReplicationGroup(input)
+	if err != nil {
+		return fmt.Errorf("error creating ElastiCache Global Replication Group: %w", err)
+	}
+
+	d.SetId(aws.StringValue(output.GlobalReplicationGroup.GlobalReplicationGroupId))
+
+	if _, err := waiter.GlobalReplicationGroupAvailable(conn, d.Id(), waiter.GlobalReplicationGroupDefaultCreatedTimeout); err != nil {
+		return fmt.Errorf("error waiting for ElastiCache Global Replication Group (%s) availability: %w", d.Id(), err)
+	}
+
+	return resourceAwsElasticacheGlobalReplicationGroupRead(d, meta)
+}
+
+func resourceAwsElasticacheGlobalReplicationGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).elasticacheconn
+
+	globalReplicationGroup, err := finder.GlobalReplicationGroupByID(conn, d.Id())
+	if !d.IsNewResource() && tfresource.NotFound(err) {
+		log.Printf("[WARN] ElastiCache Global Replication Group (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("error reading ElastiCache Replication Group: %w", err)
+	}
+
+	if aws.StringValue(globalReplicationGroup.Status) == "deleting" || aws.StringValue(globalReplicationGroup.Status) == "deleted" {
+		log.Printf("[WARN] ElastiCache Global Replication Group (%s) in deleted state (%s), removing from state", d.Id(), aws.StringValue(globalReplicationGroup.Status))
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("arn", globalReplicationGroup.ARN)
+	d.Set("at_rest_encryption_enabled", globalReplicationGroup.AtRestEncryptionEnabled)
+	d.Set("auth_token_enabled", globalReplicationGroup.AuthTokenEnabled)
+	d.Set("cache_node_type", globalReplicationGroup.CacheNodeType)
+	d.Set("cluster_enabled", globalReplicationGroup.ClusterEnabled)
+	d.Set("engine", globalReplicationGroup.Engine)
+	d.Set("actual_engine_version", globalReplicationGroup.EngineVersion)
+	d.Set("global_replication_group_description", globalReplicationGroup.GlobalReplicationGroupDescription)
+	d.Set("global_replication_group_id", globalReplicationGroup.GlobalReplicationGroupId)
+	d.Set("transit_encryption_enabled", globalReplicationGroup.TransitEncryptionEnabled)
+
+	d.Set("primary_replication_group_id", flattenElasticacheGlobalReplicationGroupPrimaryGroupID(globalReplicationGroup.Members))
+
+	if err := d.Set("global_replication_group_members", flattenElasticacheGlobalReplicationGroupMembers(globalReplicationGroup.Members)); err != nil {
+		return fmt.Errorf("error setting global_cluster_members: %w", err)
+	}
+
+	return nil
+}
+
+func resourceAwsElasticacheGlobalReplicationGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).elasticacheconn
+
+	// Only one field can be changed per request
+	updaters := map[string]elasticacheGlobalReplicationGroupUpdater{}
```

Comment:
Would recommend refactoring this "clever" logic in preference of simpler per attribute checks for consistency with the rest of the provider, especially in the future when attribute change ordering might matter. 👍

Reply:
This was left over from deferred work to that allows updating other fields on the resource